### PR TITLE
feat(router): added dispute retrieve and dispute list apis

### PR DESCRIPTION
**Pull request by sai-harsha-vardhan**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description
Added support for the following APIs
1. Disputes - Retrieve => Retrieves the dispute based on dispute_id
2. Disputes - List => Lists disputes for a merchant based on some filters like dispute_stage, dispute_status etc.

### Additional Changes

- [x] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?
Tested Manually
**Retrieve Dispute API**
Success - 200
image

Failure (Not Found) - 404
image

**List Disputes API**
image

Using filters
image

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: migrations/2023-03-15-185959_add_dispute_table/up.sql [hunk 1]
```diff
@@ -23,7 +23,7 @@ CREATE TABLE dispute (
     modified_at TIMESTAMP NOT NULL DEFAULT now()::TIMESTAMP
 );
 
-CREATE UNIQUE INDEX dispute_id_index ON dispute (dispute_id);
+CREATE UNIQUE INDEX merchant_id_dispute_id_index ON dispute (merchant_id, dispute_id);
```
Comment (vspecky) [inline migrations/2023-03-15-185959_add_dispute_table/up.sql:26]: Check whether a change is required in `down.sql` as well.

**Review by vspecky — CHANGES_REQUESTED** [review]

File: migrations/2023-03-15-185959_add_dispute_table/up.sql (hunk 1 above)
Comment (sai-harsha-vardhan) [reply migrations/2023-03-15-185959_add_dispute_table/up.sql:26]: We are dropping `dispute` table in down.sql, so are indices!

**Review by Narayanbhat166 — APPROVED** [review]
LGTM

File: crates/router/src/core/disputes.rs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+) -> RouterResponse<Vec<api_models::disputes::DisputeResponse>> {
+    let disputes = state
+        .store
+        .find_disputes_by_merchant_id(&merchant_account.merchant_id, constraints)
+        .await
+        .map_err(|error| {
+            error.to_not_found_response(errors::ApiErrorResponse::InternalServerError)
+        })?;
+    let mut disputes_list: Vec<api_models::disputes::DisputeResponse> = vec![];
+    for dispute in disputes {
```
Comment (jarnura) [inline crates/router/src/core/disputes.rs:43]: use map instead

File: crates/router/src/routes/app.rs [hunk 3]
```diff
@@ -380,6 +382,18 @@ impl ApiKeys {
     }
 }
 
+pub struct Disputes;
+
+#[cfg(any(feature = "olap", feature = "oltp"))]
+impl Disputes {
+    pub fn server(state: AppState) -> Scope {
+        web::scope("/disputes")
```
Comment (jarnura) [inline crates/router/src/routes/app.rs:390]: add olap feature to the routes

**Review by jarnura — CHANGES_REQUESTED** [review]

**Review by jarnura — APPROVED** [review]

**Review by Narayanbhat166 — APPROVED** [review]

**Pull request merged**